prusti-common/src/vir/optimizations/folding/expressions.rs

```diff
@@ -242,23 +242,18 @@ fn find_common_unfoldings2(
 /// Find unfoldings that are in all three sets.
 fn find_common_unfoldings3<'a>(
     mut first: UnfoldingMap,
-    mut first_reqs: &'a RequirementSet,
+    mut _first_reqs: &'a RequirementSet,
     mut second: UnfoldingMap,
     mut second_reqs: &'a RequirementSet,
     mut third: UnfoldingMap,
     third_reqs: &'a RequirementSet,
 ) -> (UnfoldingMap, UnfoldingMap, UnfoldingMap, UnfoldingMap) {
     let mut common = HashMap::new();
     let mut new_first = HashMap::new();
-    let swap = first.is_empty();
-    if swap {
-        mem::swap(&mut first, &mut second);
-        mem::swap(&mut first_reqs, &mut second_reqs);
-    }
     for (place, data) in first {
-        let second_agrees = second.contains_key(&place) ||
+        let second_agrees = second.contains_key(&place); ||
             second_reqs.iter().all(|p| !p.has_prefix(&place));
-        let third_agrees = third.contains_key(&place) ||
+        let third_agrees = third.contains_key(&place); ||
             third_reqs.iter().all(|p| !p.has_prefix(&place));
```

Comment:
This looks like a syntax error, but perhaps it is interpreted as a closure after the semicolon?

Reply:
Yes, that's certainly a typo.